nodejs/aws-serverless/cloudwatch/logGroup.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import { cloudwatch, lambda } from "@pulumi/aws";
+import { LogGroup } from "@pulumi/aws/cloudwatch";
+import * as pulumi from "@pulumi/pulumi";
+import { RunError } from "@pulumi/pulumi";
+import { createLambdaFunction, Handler } from "./../function";
+import { EventSubscription } from "./../subscription";
+
+/**
+ * Arguments to control the event rule subscription.  Currently empty, but still defined in case of
+ * future need.
+ */
+interface LogGroupEventSubscriptionArgs {
+    /**
+     * A valid CloudWatch Logs filter pattern for subscribing to a filtered stream of log events.
+     */
+    filterPattern?: string;
+}
+
+interface LogGroupEvent {
+    awslogs: {
+        // The data attribute in the Lambda record is Base64 encoded and compressed with the gzip
+        // format.
+        data: string;
+    };
+}
+
+interface DecodedLogGroupEvent {
```

Comment:
@lukehoban Here's another interesting case.  cloudwatch sends log-group events that are pretty sucky.  They're basically just ```data: "base64, gzip compressed"```.  To use them, you have to remove the base64 decoding, then decompress.  I expose a helper here to let you do that: decodeLogGroupEvent.  So you would do:

```ts
onEvent(name, async (encoded) =>  {
     const decoded = await logGroup.decodeLogGroupEvent(encoded);
     // use decoded
});
```

But, perhaps we should just have another way to register that does this work for you.  i.e. 

```ts
onDecodedEvent(name, async (decoded) =>  {
     // use decoded
});
```

?

Of course, we could always add that later.